backend/src/routes/courses.js

```diff
@@ -2,28 +2,50 @@ const express = require('express');
 const router = express.Router();
 const db = require('../db');
 const datefns = require('date-fns');
+const utils = require('../utils');
 
-router.get('/', async (req, res) => {
-    const timestampToDate = (date) => datefns.parse(Number(date), 'MM-DD-YYYY');
-    const toFormattedDate = (date) => datefns.format(date);
+const getCourses = async (req, res) => {
+    try {
+        const timestampToDate = (date) =>
+            datefns.parse(Number(date), 'MM-DD-YYYY');
+        const toFormattedDate = (date) => datefns.format(date);
 
-    // Default start and end date for the last week
-    const startDate = toFormattedDate(
-        req.query.startDate ? timestampToDate(req.query.startDate) : new Date()
-    );
-    const endDate = toFormattedDate(
-        req.query.endDate
-            ? timestampToDate(req.query.endDate)
-            : datefns.addWeeks(new Date(), 1)
-    );
+        // Default start and end date for the last week
+        const startDate = toFormattedDate(
+            req.query.startDate
+                ? timestampToDate(req.query.startDate)
+                : new Date()
+        );
+        const endDate = toFormattedDate(
+            req.query.endDate
+                ? timestampToDate(req.query.endDate)
+                : datefns.addWeeks(new Date(), 1)
+        );
 
-    const courses = await db.courses.getCourses(startDate, endDate);
-    const response = db.courses.reduceCoursesByDate(courses);
-    res.status(200).send(response);
-});
+        const courses = await db.courses.getCourses(startDate, endDate);
+        const response = db.courses.reduceCoursesByDate(courses);
+        res.status(200).send(response);
+    } catch (err) {
+        res.status(500).send(`Failed to get courses. Error: ${err.message}`);
+    }
+};
 
-router.get('/:id', (req, res) => {
-    res.send('returning singe course ' + req.params.id);
-});
+const getCourseById = async (req, res) => {
+    try {
+        const courseId = req.params.id;
+        const validationErrors = utils.courses.validateCourseId(courseId);
+        if (validationErrors) {
+            res.status(420).send(validationErrors);
+        } else {
+            const course = await db.courses.getCourseById(courseId);
```

Comment:
What will happen now if there is no course with that id? Should we handle the 404 case somehow?